packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -664,12 +687,46 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
         }
     };
 
+    importFieldsFromJson = (file: File): Promise<SimpleResponse> => {
+        const { domain, onChange } = this.props;
+
+        return file.text()
+            .then(text => {
+                const domainType = domain.domainType;
+                const jsFields = JSON.parse(text);
+                if (jsFields.length < 1) return {success: false, msg: 'No fields found.'};
+
+                for (let i=0; i < jsFields.length; i++){
+                    let field = jsFields[i];
+
+                    if (field.defaultValueType && !domain.hasDefaultValueOption(field.defaultValueType)) {
+                        return {success: false, msg: `Error on importing field '${field.name}': Default value '${field.defaultValueType}' is invalid.`};
+                    }
+
+                    if (domainType !== 'list' && field.lockType === DOMAIN_FIELD_PRIMARY_KEY_LOCKED) {
+                        return {success: false, msg: `Error on importing field '${field.name}': Domain type '${domainType}' does not support fields with an externally defined Primary Key.`};
+                    }
+                }
+
+                // Convert to TS and merge entire List
+                const tsFields: List<DomainField> = List(jsFields.map(field => DomainField.create(field, true)));
```

Comment:
There are likely some properties of a field that we should either exclude from the export or drop during the import, namely `propertyId` and `propertyURI` as those are specific values set on the server side during save. I believe these values are null/undefined when you click the "Add Field" button as well so we'll want to match that for the import case.

Reply:
Ah, gotcha. I've added this in the error-checking section above this section of the code. Confirmed that with a new field, `propertyId` and `propertyURI` are non-existent on field object.

Done!